### PR TITLE
mgmt, support additional methods for appplatform deployment

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Changed `cpu` and `memory` from String to Double.
Support `jvmOptions`, `runtimeVersion` in `SpringAppDeployment`.
Add `Java_17` to `RuntimeVersion` (confirmed with service team, they support this in all tiers now).

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java [hunk 1]
```diff
@@ -175,6 +175,56 @@ public List<String> configFilePatterns() {
...
+    }
+
+    @Override
+    public Double memoryInGB() {
+        return Utils.fromMemoryString(innerModel().properties().deploymentSettings().resourceRequests().memory());
+    }
+
+    @Override
+    public RuntimeVersion runtimeVersion() {
+        if (isEnterpriseTier() || innerModel().properties() == null) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java:190]: Does EnterpriseTier tell us nothing?

If really so, document it.

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/models/SpringAppDeployment.java [hunk 2]
```diff
@@ -182,10 +194,10 @@ interface DefinitionStages {
 
             /**
              * Specifies the cpu number of the deployment.
-             * @param cpuCount the number of the cpu, 1 core can be represented by 1 or 1000m
+             * @param cpuCount the number of the cpu, can be 0.5, 1, 2, etc
              * @return the next stage of deployment definition
              */
-            T withCpu(String cpuCount);
+            T withCpu(double cpuCount);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/models/SpringAppDeployment.java:200]: Good is easy to use, bad is service is not obligated to keep the same format within api-version (though not likely to change).

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java:190]: No, currently they can't give us the deduced version. Will add docs.

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        } else {
+            return String.format("%dMi", memoryDecimal.multiply(D_1024).intValue());
+        }
+    }
+
+    private static boolean isInteger(double sizeInGB) {
+        return (sizeInGB % 1) == 0;
+    }
+
+    private static class CpuMatcher {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java:92]: Please add test cases for these (package private access should be fine).

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        String getFraction() {
+            return this.fractionMatcher.group(1);
+        }
+    }
+
+    private static class MemoryMatcher {
+        static final Pattern MEMORY_GB = Pattern.compile("^([0-9]+)Gi"); // 1Gi, 2Gi
+        static final Pattern MEMORY_MB = Pattern.compile("^([0-9]+)Mi"); // 512Mi, 1024Mi
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java:127]: Try double check with service that they will honor case (e.g. they won't change return value from 512Mi to 512mi or 512MI).

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java (hunk 4 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java:127]: From the regex they give:
@"^(?<number>0|-?[1-9][0-9]*)(?<unit>Gi|Mi)$ for mem 
@"^(?<number>0|-?[1-9][0-9]*)(?<unit>m?)$ for cpu

It is case-sensitive.

File: sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java (hunk 3 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java:92]: Added.

**Pull request merged**